pkg/bootstrap/option/convert.go

```diff
@@ -180,6 +181,33 @@ func durationConverter(value *types.Duration) convertFunc {
 	}
 }
 
+// openCensusAgentContextConverter returns a converter that returns the list of
+// distributed trace contexts to propagate with envoy.
+func openCensusAgentContextConverter(contexts []meshAPI.Tracing_OpenCensusAgent_TraceContext) convertFunc {
+	allContexts := `["TRACE_CONTEXT","GRPC_BIN","CLOUD_TRACE_CONTEXT","B3"]`
+	return func(*instance) (interface{}, error) {
+		if len(contexts) == 0 {
+			return allContexts, nil
+		}
+
+		var envoyContexts []string
+		for _, c := range contexts {
+			switch c {
+			// Ignore UNSPECIFIED
+			case meshAPI.Tracing_OpenCensusAgent_W3C_TRACE_CONTEXT:
+				envoyContexts = append(envoyContexts, "TRACE_CONTEXT")
+			case meshAPI.Tracing_OpenCensusAgent_GRPC_BIN:
+				envoyContexts = append(envoyContexts, "GRPC_BIN")
+			case meshAPI.Tracing_OpenCensusAgent_CLOUD_TRACE_CONTEXT:
+				envoyContexts = append(envoyContexts, "CLOUD_TRACE_CONTEXT")
+			case meshAPI.Tracing_OpenCensusAgent_B3:
+				envoyContexts = append(envoyContexts, "B3")
```

Comment:
Should there be a default error case in switch or by construction there cannot be values other than the ones you’re switching on?

Reply:
By construction there cannot be more, aside from Tracing_OpenCensusAgent_UNSPECIFIED, which is intentionally skipped.